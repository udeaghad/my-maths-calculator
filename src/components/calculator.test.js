import {render, fireEvent} from '@testing-library/react';
import CreateCalculator from './Calculator';
import calculate from './logic/calculate';
import operate from './logic/operate'

describe('To check calculator buttons', () => { 
test('To make sure the number of buttons are complete', () =>{   
        const{ getAllByRole} = render(<CreateCalculator />)
        const button = getAllByRole('button')
        expect(button).toHaveLength(19)   
})
test('To check if correct value is displayed on button click', () => {
    
        const {getByTestId} = render(<CreateCalculator />);
        const showResult= getByTestId('display-result');
        const btnClick = getByTestId('btn-7')       
        let getValue = btnClick.innerHTML
        fireEvent.click(btnClick, {target:{innerText:getValue}});
        expect(showResult.textContent).toContain('7')

    })
})

describe('To test calculator logic functions', () => {
    test('To test calculate function', () => {
    
    let buttonName = '='

    const obj = {
        total: 10,
        next: 5,
        operation: '+'
    }

    let totalValue = (calculate(obj, buttonName))

   const {total} = totalValue

    expect(total).toBe('15')

    })

    test('To test the operate fuctions', () => {
        const obj = {
            total: 10,
            next: 5,
            operation: 'x'
        }

        const{total, next, operation} = obj 

        let totalValue = operate(total, next, operation)
        

        expect(totalValue).toBe('50')
        
    })
})

describe ('To do snapshot of the React component', () => {
  test('To do snapshot of create calculate component', () => {
    const tree = render.create(<CreateCalculator />).JSON();
    expect(tree).toMatchSnapshot()
  })
})