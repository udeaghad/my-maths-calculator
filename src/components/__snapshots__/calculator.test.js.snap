// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To do snapshot of the React component To do snapshot of CreateCalculate component 1`] = `
<div
  className="container"
>
  <h3>
    Lets do some maths!
  </h3>
  <div
    className="calc-container"
  >
    <div
      className="show-result"
      data-testid="display-result"
    >
      0
    </div>
    <div
      className="special-fxn"
    >
      <button
        className="btn-fxn"
        data-testid="btn-AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-fxn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-fxn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="digit-btn"
    >
      <button
        className="btn"
        data-testid="btn-7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="zero-dot"
    >
      <button
        className="zero-btn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="dot-btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="operators"
    >
      <button
        className="op-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="op-btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="op-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="op-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="op-btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`To do snapshot of the React component To do snapshot of HomePage component 1`] = `
<div
  className="HomeContainer"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer porta tempor felis eu mollis. Mauris eget tristique massa. Nullam id mi bibendum, pretium augue in, venenatis metus. Mauris elit mauris, accumsan eget ultricies eget, consequat vel enim. Nam vitae nisl accumsan, ornare nulla a, congue diam. Donec hendrerit turpis nulla, pulvinar scelerisque sapien luctus a. sodales sodales.
  </p>
  <br />
  <p>
    Quisque ac ultrices nunc. Curabitur risus augue, hendrerit at nulla in, aliquet tempus augue. Mauris tincidunt neque id orci eleifend varius.Nullam at lacus facilisis, congue massa ut, condimentum velit. Fusce ac suscipit nulla. Nulla condimentum turpis non mollis maximus. Ut sodales risus eget lorem suscipit tempus. Nulla sed sagittis nulla. Interdum et malesuada fames ac ante ipsum primis in faucibus.Cras non molestie est, eu consectetur ligula.
  </p>
</div>
`;

exports[`To do snapshot of the React component To do snapshot of QuotePage component 1`] = `
<div
  className="quoteContainer"
>
  <h3>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
    <br />
    (William Paul Thurston)10 Jun 2014
  </h3>
</div>
`;
